### PR TITLE
HTTP API

### PR DESCRIPTION
**Pull request by penguinpowernz**
Hiya, just noticed you've been working on the repo, and thought I'd make this PR to see what has changed.  However I should delete it and pull your new stuff into my repo, resolve the conflicts and make a new one.  But would be interested in your comments on the changes I've made if you have any/have time. :)

### CODE REVIEW

**Comment by opencoff** [comment]
Hi! Thank you for taking an interest in the project.

At a high level - I wrote ovpn-tool as a command line tool to help me (and others I know) to manage their openvpn installation. I am unsure what problem you are trying to address by turning it into a daemon/server. Perhaps you can explain the user-journey  that you want to accomplish?

**Comment by penguinpowernz** [comment]
Oh yea so basically I wanted to be able to do what the tool does but via an HTTP API rather than CLI. This has helped me to integrate the tool with a rails app that manages VPN connected devices.

However if you would not like it to go into mainline, then it can be independent because really it shells out from the API call and calls the tools binary.  I did need to just allow specifying the password in a envvar similar to the way that I saw [github.com/restic/restic](github.com/restic/restic) doing it with repository passwords - thus removing the need to use the DB password everytime the command is run, kind of like caching it.

File: src/init.go [hunk 1]
```diff
@@ -55,24 +55,32 @@ func initCA(dbfile string, args []string, init bool) *pki.CA {
 	var pw string
 
 	args = fs.Args()
-	if len(args) > 0 {
-		cn = args[0]
-		creat = true
-
-		pw, err = utils.Askpass("Enter password for DB", true)
+	if !init && os.Getenv("PASSWD_FILE") != "" {
```
Comment (penguinpowernz) [inline src/init.go:58]: This is the line that reads the password out of the file specified in an envvar

File: src/export.go [hunk 2]
```diff
@@ -383,11 +383,12 @@ script-security 2
...
-remote-cert-tls server
+# when a ECDSA key is used then "keyAgreement" flag is needed for being ECDH "capable" (as opposed to ephemeral ECDHE)
+remote-cert-ku a8 a0 80 88
 tls-version-min 1.2
 
 # Verify the remote server's common name
-verify-x509-name "{{ .ServerCommonName }}"
+verify-x509-name "{{ .ServerCommonName }}" name
 
 # Inline certs, keys and tls-crypt follows
```
Comment (penguinpowernz) [inline src/export.go:393]: I also had problems connecting to the server without these options, not sure if you ever came across the same bug or already fixed it.

**Comment by penguinpowernz** [comment]
Sorry I probably should have given you smaller targetted commits rather than just dumping this larger cognitive load.

**Pull request closed without merging**